plugins/modules/iparole.py

```diff
@@ -285,8 +264,9 @@ def ensure_role_with_members_is_present(module, name, res_find, action):
             del_members[key] = [to_text(item) for item in del_list]
 
     service = [
-        to_text(svc) if '@' in svc else ('%s@%s' % (svc, api_get_realm()))
-        for svc in (module_params_get(module, 'service') or [])
+        to_text(svc) if '@' in svc else ('%s@%s' % (svc,
+                                                    module.ipa_get_realm()))
+        for svc in (module.params_get('service') or [])
```

Comment:
I like the format suggestion by `black` for this code snippet:
```python
    to_text(svc)
    if "@" in svc
    else ("%s@%s" % (svc, module.ipa_get_realm()))
    for svc in (module.params_get("service") or [])
```
I find it easier to read.

Reply:
What do you think about using `afm_convert` for all methods in IPAAnsibleModule. This way we could get rid of all `to_text` calls in the modules.

Reply:
No having to use `to_text` on the modules would be nice. Sometimes this errors show up very late.

Reply:
Let's test the integration of _afm_convert and to_text in another PR. It might take some time to get all sorted out.